src/System.Net.HttpListener/src/System/Net/HttpRequestStream.cs

```diff
@@ -65,6 +65,19 @@ public override IAsyncResult BeginRead(byte[] buffer, int offset, int size, Asyn
 
             return BeginReadCore(buffer, offset, size, callback, state);
         }
+        
+        public new Task<int> ReadAsync(Byte[] buffer, int offset, int count)
+        {
+            return ReadAsync(buffer, offset, count, CancellationToken.None);
+        }
+        
+        public override Task<int> ReadAsync(byte[] buffer, int offset, int size, CancellationToken cancellationToken)
+        {
+            return Task<int>.Factory.FromAsync(
+                (callback, state) => this.BeginRead(buffer, offset, size, callback, state),
+                iar => this.EndRead(iar),
```

Comment:
The above two lines are going to create unnecessary delegate and closure allocations on each call. The delegates will be cached and closure allocations avoided by writing it as:

```c#
return Task.Factory.FromAsync(
    (localBuffer, localOffset, localCount, callback, state) => ((Stream)state).BeginRead(localBuffer, localOffset, localCount, callback, state),
    iar => ((Stream)iar.AsyncState).EndRead(iar),
    buffer, offset, count, this);
```